OCPBUGS-10304: Document oc-mirror command for creating ImageContentSou… (#63332)
openshift/openshift-docs: modules/installation-initializing-manual.adoc

You must name this configuration file `install-config.yaml`.
====
ifdef::restricted[]

** Unless you use a registry that {op-system} trusts by default, such as
`docker.io`, you must provide the contents of the certificate for your mirror
repository in the `additionalTrustBundle` section. In most cases, you must
provide the certificate for your mirror.
** You must include the `imageContentSources` section from the output of the command to
mirror the repository.

> ** The 'ImageContentSourcePolicy' is generated as an output of `oc-mirror` flow (after the mirroring process is finished).

Reviewer: ImageContentSourcePolicy was [deprecated in 4.13][1].  Is that still what `oc-mirror` is generating in 4.14?

[1]: https://docs.openshift.com/container-platform/4.13/release_notes/ocp-4-13-release-notes.html#ocp-4-13-icsp

Author: Thanks for that input, @wking ! I'm using the information from the jira. Would it work if I indicated in a NOTE that ImageContentSourcePolicy is not available in 4.13+?

Author: I'm glad you chimed in, I need review of this PR so I can get it completed and closed out!

Author: @wking there's a section in the documentation that describes the issues with ImageContentSourcePolicy. I'm referring to that in the note. please PTAL!

> ** The 'ImageContentSourcePolicy' is generated as an output of `oc-mirror` flow (after the mirroring process is finished).

Reviewer 2: ```suggestion
** The 'ImageContentSourcePolicy' is generated as an output of `oc mirror` flow after the mirroring process is finished.
```

Author: So, this is interesting. The downloadable module is called "oc-mirror". It provides the command "oc mirror".

> ** `oc-mirror` generates the file called 'ImageContentSourcePolicy' which contains yaml needed to define 'ImageContentSourcePolicy'. 

Reviewer 2: ```suggestion
** The `oc mirror` command generates a 'ImageContentSourcePolicy' file, which contains the YAML needed to define 'ImageContentSourcePolicy'. 
```

Author: Done, thank you!

> Copy the text from this file and paste in your install-config.yaml. 

Reviewer 2: ```suggestion
Copy the text from this file and paste it into your `install-config.yaml` file. 
```

Author: Done, thank you!

> ** You must run 'oc-mirror' twice. The first time you run `oc-mirror` you will get a full ImageContentSourcePolicy

Reviewer 2: I rewrote this section. My main suggestion is here is not to use language like "Please". Do not use please in any instructions. Terms of politeness are superfluous, convey the wrong tone for technical material, and are not regarded the same way in all cultures.

```suggestion
** You must run the 'oc mirror' command twice. The first time you run the`oc mirror` command, you get a full `ImageContentSourcePolicy` file. The second time you run the `oc mirror command`, you only get the difference between the first and second run. Because of this behavior, you must always keep a backup of these files in case you need to merge them into one, complete `ImageContentSourcePolicy` file. Keeping a backup of these two files ensures that you have everything you need for a complete `ImageContentSourcePolicy` file.
```

Author: I didn't write that section but I MUST agree with you. Glad to make the changes!

====
You must name this configuration file `install-config.yaml`.
====

ifdef::restricted,restricted-upi[]

** Unless you use a registry that {op-system} trusts by default, such as
`docker.io`, you must provide the contents of the certificate for your mirror
repository in the `additionalTrustBundle` section. In most cases, you must
provide the certificate for your mirror.
** You must include the `imageContentSources` section from the output of the command to
mirror the repository.

> 

Reviewer 3: Need a `+` here

> 

Reviewer 3: Need a `+` before the admonition

Author: Done

[IMPORTANT]
====

> ** The 'ImageContentSourcePolicy' is generated as an output of `oc mirror` after the mirroring process is finished.

Reviewer 3: ```suggestion
** The 'ImageContentSourcePolicy' file is generated as an output of `oc mirror` after the mirroring process is finished.
```

Author: Done

Author: Done

> ** The `oc mirror` command generates a ImageContentSourcePolicy' file which contains the YAML needed to define 'ImageContentSourcePolicy'. 

Reviewer 3: The second _ImageContentSourcePolicy_ needs a noun. I see the term [resource](https://github.com/openshift/openshift-docs/pull/63332/files#diff-57166dbcd99d39b77ac269edbafbd33682c7a4ef400cf2fc3f12fe7b2c3cb57dR42) below.
```suggestion
** The `oc mirror` command generates an 'ImageContentSourcePolicy' file, which contains the YAML needed to define the 'ImageContentSourcePolicy' resource. 
```

Copy the text from this file and paste it into your `install-config.yaml` file. 

> ** You must run the 'oc mirror' command twice. The first time you run the `oc mirror` command, you get a full `ImageContentSourcePolicy` file. The second time you run the `oc mirror` command, you only get the difference between the first and second run. 

Reviewer 3: After which run do you copy the text from the ICSP file?

Reviewer 3: ```suggestion
** You must run the `oc mirror` command twice. The first time you run the `oc mirror` command, you get a full `ImageContentSourcePolicy` file. The second time you run the `oc mirror` command, you only get the difference between the first and second run. 
```
